src/node/SetupServerApi.ts

```diff
@@ -63,13 +72,24 @@ export class SetupServerApi
       // If in remote mode, await the WebSocket connection promise
       // before handling any requests.
       if (this.resolvedOptions.remotePort) {
-        /**
-         * @fixme This will never resolve.
-         * Playwright runs the app first (this part) and THEN it runs
-         * the tests (setupRemoteServer part). This causes the "/"
-         * request to pend forever on this line since the app has to
-         * load successfully before the tests run.
-         */
+        const remoteResolutionUrl = createWebSocketServerUrl(
+          this.resolvedOptions.remotePort,
+        )
+        // Build a pattern to match the socket.io connection
+        remoteResolutionUrl.pathname = '/socket.io'
```

Comment:
The pathname here depends on socket-io's default value, because MSW doesn't set the value in `opts.path` (see `setupRemoteServer.ts`, line 210, second arg). 
If you preferred, then that location could be changed to set the path, and then this wouldn't be dependent on their potential changes to default values.

Reply:
Moved to L89 in SetupServerCommonApi.ts, resolving here and copying to new location (changed by merging v2.2.0).